src/github.com/getlantern/balancer/balancer_test.go

```diff
@@ -191,13 +193,16 @@ func TestCheck(t *testing.T) {
 	assert.NoError(t, err)
 
 	// recheck failed dialer
+	before := atomic.LoadUint32(&checkCount)
+	// Can't rely on WaitGroup to exactly match the check count. It's time sensitive.
+	// Adds a large enough delta to avoid wg.Done() panicking.
```

Comment:
Oh, I see.  I wonder if there's a safer way to model this, because when I run the test with the `-race` flag on Linux 64 bit, I see a panic.

```
DEBUG balancer: balancer.go:106 Dialing tcp://does-not-exist.com:80 with 
DEBUG balancer: balancer.go:112 Successfully dialed via  to tcp://does-not-exist.com:80 on pass 0
DEBUG balancer: balancer.go:79 Balancer idle for 783.053ms, start checking all dialers
panic: sync: negative WaitGroup counter

goroutine 3135 [running]:
panic(0x8398c0, 0xc82034a070)
	/usr/local/go/src/runtime/panic.go:481 +0x3ff
sync.(*WaitGroup).Add(0xc8201b80c0, 0xffffffffffffffff)
	/usr/local/go/src/sync/waitgroup.go:71 +0x164
sync.(*WaitGroup).Done(0xc8201b80c0)
	/usr/local/go/src/sync/waitgroup.go:96 +0x38
github.com/getlantern/balancer.TestCheck.func3(0xc820402120)
	/home/ox/git/lantern/src/github.com/getlantern/balancer/balancer_test.go:171 +0x56
github.com/getlantern/balancer.(*dialer).check(0xc82013e030)
	/home/ox/git/lantern/src/github.com/getlantern/balancer/dialer.go:92 +0x19d
created by github.com/getlantern/balancer.(*Balancer).checkDialers
	/home/ox/git/lantern/src/github.com/getlantern/balancer/balancer.go:131 +0xc7
FAIL	github.com/getlantern/balancer	3.658s
```